server/chat-plugins/datasearch.ts

```diff
@@ -1709,48 +1874,47 @@ function runLearn(target, cmd) {
 		return {error: "You must specify at least one move."};
 	}
 
-	let lsetProblem;
-	let moveNames = [];
+	let lsetProblem: {[k: string]: any, type: string, moveName: string} = Object.create(null);
```

Comment:
There was a crash surrounding this in my initial commit; I fixed it, but now running /learn just always results in saying the Pokemon cant learn a move, even if they actually can. I'm honestly not sure how to fix this.

Reply:
I've managed to fix this but the code could probably be optimized